app/javascript/pages/Wishlists/Index.tsx

```diff
@@ -17,54 +14,60 @@ import { WithTooltip } from "$app/components/WithTooltip";
 
 import placeholder from "$assets/images/placeholders/wishlists.png";
 
-type Wishlist = {
+export type Wishlist = {
   id: string;
   name: string;
   url: string;
   product_count: number;
   discover_opted_out: boolean;
 };
 
-const WishlistsPage = ({
-  wishlists: preloadedWishlists,
-  reviews_page_enabled,
-  following_wishlists_enabled,
-}: {
+type Props = {
   wishlists: Wishlist[];
   reviews_page_enabled: boolean;
   following_wishlists_enabled: boolean;
-}) => {
+};
+
+const WishlistsPage = ({ wishlists: preloadedWishlists, reviews_page_enabled, following_wishlists_enabled }: Props) => {
   const [wishlists, setWishlists] = React.useState<Wishlist[]>(preloadedWishlists);
   const [deletingWishlist, setConfirmingDeleteWishlist] = React.useState<Wishlist | null>(null);
   const [isDeleting, setIsDeleting] = React.useState(false);
 
   const destroy = async (id: string) => {
     setIsDeleting(true);
 
-    try {
-      await deleteWishlist({ wishlistId: id });
-      setWishlists(wishlists.filter((wishlist) => wishlist.id !== id));
-      setConfirmingDeleteWishlist(null);
-      showAlert("Wishlist deleted!", "success");
-    } catch (e) {
-      assertResponseError(e);
-      showAlert("Sorry, something went wrong. Please try again.", "error");
-    } finally {
-      setIsDeleting(false);
-    }
+    router.delete(Routes.wishlist_path(id), {
+      preserveScroll: true,
+      onSuccess: () => {
+        setWishlists(wishlists.filter((wishlist) => wishlist.id !== id));
+        setConfirmingDeleteWishlist(null);
+        showAlert("Wishlist deleted!", "success");
+      },
+      onError: () => showAlert("Sorry, something went wrong. Please try again.", "error"),
+      onFinish: () => setIsDeleting(false),
+    });
   };
 
   const updateDiscoverOptOut = async (id: string, optOut: boolean) => {
-    try {
-      setWishlists(
-        wishlists.map((wishlist) => (wishlist.id === id ? { ...wishlist, discover_opted_out: optOut } : wishlist)),
-      );
-      await updateWishlist({ id, discover_opted_out: optOut });
-      showAlert(optOut ? "Opted out of Gumroad Discover." : "Wishlist is now discoverable!", "success");
-    } catch (e) {
-      assertResponseError(e);
-      showAlert("Sorry, something went wrong. Please try again.", "error");
-    }
+    setWishlists(
+      wishlists.map((wishlist) => (wishlist.id === id ? { ...wishlist, discover_opted_out: optOut } : wishlist)),
+    );
+
+    router.put(
+      Routes.wishlist_path(id),
+      { wishlist: { discover_opted_out: optOut } },
+      {
+        preserveScroll: true,
+        onSuccess: () =>
+          showAlert(optOut ? "Opted out of Gumroad Discover." : "Wishlist is now discoverable!", "success"),
```

Comment:
This should be handled server side as a flash message, since we intercept them in the layout.